src/lightning/pytorch/callbacks/progress/tqdm_progress.py

```diff
@@ -262,6 +264,8 @@ def on_train_start(self, *_: Any) -> None:
 
     @override
     def on_train_epoch_start(self, trainer: "pl.Trainer", *_: Any) -> None:
+        if self._leave:
+            self.train_progress_bar = self.init_train_tqdm()
```

Comment:
Instead of doing this, I believe it would be better to pass `leave=self.leave` to the tqdm bar directly (see `init_train_tqdm()`, `init_validation_tqdm()` etc. above.

Reply:
No, I'm saying simply pass `leave=self.leave` in `init_train_tqdm()` where we hard coded it to True so far. Line 205